lib/active_storage/service/db_service_rails70.rb

```diff
@@ -20,7 +20,8 @@ def compose(source_keys, destination_key, **)
     def current_host
       opts = url_options || {}
       url = "#{opts[:protocol]}#{opts[:host]}"
-      url + ":#{opts[:port]}" if opts[:port]
+      url += ":#{opts[:port]}" if opts[:port]
```

Comment:
What about using `<<` instead?

```suggestion
      url << ":#{opts[:port]}" if opts[:port]
```

Reply:
That is not a good idea while you still support ruby 2.7. Ruby 3 updated the frozen string literal setting to account dynamic string literals, but ruby 2.7 will error `can't modify frozen String` for anyone that has turned that on globally.

Reply:
Sorry for the late reply.

`+=` duplicates the string (you can make a test with `object_id`)

You can setup an unfrozen string when you are preparing it:
`url = +"#{opts[:protocol]}#{opts[:host]}"`